ehrql/query_engines/mssql.py

```diff
@@ -151,27 +153,86 @@ def create_inline_table(self, columns, rows):
         ]
         return table
 
+    def get_measure_queries(self, grouped_sum, results_query):
+        """
+        Return the SQL queries to fetch the results for a GroupedSum representing
+        a collection of measures that share a denominator.
+        A GroupedSum contains:
+        - denominator: a single column to sum over
+        - numerator: a tuple of columns to sum over, grouped by their respective
+        - group_bys: a tuple of tuples of columns to group each numerator by
+        Uses GROUPING SETS to combine multiple group by clauses into one
+        GROUP BY, meaning that we can query all the measures in one go.
+        We add the numerator and denominator queries for all measures,
+        and then group by the grouping set for each measure.
+        A GROUPING ID on all the group by columns allows us to identify which
+        grouping level applies to each row.
+        https://learn.microsoft.com/en-us/sql/t-sql/queries/select-group-by-transact-sql?view=sql-server-ver16
+        """
+        all_sum_overs = [
+            sqlalchemy.func.sum(results_query.c[grouped_sum.denominator]).label("den")
+        ]
+        grouping_sets = []
+
+        for i, numerator in enumerate(grouped_sum.numerators):
+            all_sum_overs.append(
+                sqlalchemy.func.sum(results_query.c[numerator]).label(f"num_{i}"),
+            )
+            grouping_set = [
+                results_query.c[group_by_col]
+                for group_by_col in grouped_sum.group_bys[i]
+            ]
+            grouping_sets.append(grouping_set)
+        all_group_bys = ordered_set(iter_flatten(grouping_sets))
+        grouping_sets = [
+            sqlalchemy.tuple_(*grouping_set) for grouping_set in grouping_sets
+        ]
+        measures_query = sqlalchemy.select(
+            *all_sum_overs,
+            *all_group_bys,
+            sqlalchemy.func.grouping_id(*all_group_bys).label("grp_id"),
+        ).group_by(sqlalchemy.func.grouping_sets(*grouping_sets))
+
+        measures_results_table = temporary_table_from_query(
+            # Write the measures results to a global temporary table
+            f"##measures_results_{self.global_unique_id}",
+            measures_query,
+            index_col="num_0",
+        )
+        return [sqlalchemy.select(measures_results_table)]
```

Comment:
I _think_ that if you moved the call to `get_measure_results()` inside `get_results_queries()` as discussed somewhere above then you'd get this behaviour for free.